### PR TITLE
Install Daemonset by controller and sdk bundle update

### PR DESCRIPTION
**Pull request by mamy-CS**
- As soon as the controller is deployed before executing anything else, the controller installs daemonsets and waits for the daemonsets to be ready. With this architecture, only the controller need to be deployed and the daemonset installation is taken care of by the controller.

### CODE REVIEW

**Comment by asm582** [comment]
/hold 

until we have tests

File: internal/controller/instaslice_controller.go [hunk 1]
```diff
@@ -68,17 +70,42 @@ const requeueDelay = 2 * time.Second
 //+kubebuilder:rbac:groups=inference.codeflare.dev,resources=instaslices/finalizers,verbs=update
 //+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups="",resources=nodes,verbs=get;list;update;patch
+//+kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;delete
 
 func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 
+	// 1. Ensure DaemonSet is deployed
+	daemonSet := &appsv1.DaemonSet{}
+	err := r.Get(ctx, types.NamespacedName{Name: "instaslice-operator-controller-daemonset", Namespace: "instaslice-operator-system"}, daemonSet)
```
Comment (rphillips) [inline internal/controller/instaslice_controller.go:79]: The name and namespace should probably be constants. They are used here and in the createInstasliceDaemonset

File: internal/controller/instaslice_controller.go [hunk 2]
```diff
@@ -68,17 +70,42 @@ const requeueDelay = 2 * time.Second
...
+	if err != nil && errors.IsNotFound(err) {
+		// DaemonSet doesn't exist, so create it
+		daemonSet = createInstaSliceDaemonSet() // Add your DaemonSet creation logic here
+		err = r.Create(ctx, daemonSet)
+		if err != nil {
+			log.FromContext(ctx).Error(err, "Failed to create DaemonSet")
+			return ctrl.Result{RequeueAfter: time.Minute}, err
+		}
+		log.FromContext(ctx).Info("DaemonSet created successfully, waiting for pods to be ready")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
```
Comment (rphillips) [inline internal/controller/instaslice_controller.go:89]: This is the success scenario... RequeueAfter?

**Comment by rphillips** [comment]
Just curious, do we need to target multiarch for this operator? I saw the arm reference tag in this PR.

File: internal/controller/instaslice_controller.go (hunk 2 above)
Comment (mamy-CS) [reply internal/controller/instaslice_controller.go:91]: Correct, the reason for requeuing after creating the DaemonSet, if that's what you are asking is to allow time for the DaemonSet to initialize, schedule pods, and for those pods to start and become ready. The requeue ensures that the controller periodically checks the status of the DaemonSet after creation until it's fully deployed and all pods are ready.

File: internal/controller/instaslice_controller.go [hunk 3]
```diff
@@ -74,10 +74,8 @@ const requeueDelay = 2 * time.Second
 //+kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;delete
 
 func (r *InstasliceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-
-	emulatorMode := os.Getenv("EMULATOR_MODE")
-	log.FromContext(ctx).Info("EMULATOR_MODE ", "EMULATOR_MODE", emulatorMode)
-
+  log := logr.FromContext(ctx)
```
Comment (rphillips) [inline internal/controller/instaslice_controller.go]: seems like a formatting issue.

Comment (mamy-CS) [reply internal/controller/instaslice_controller.go]: This was taken care of by the previous pr here https://github.com/openshift/instaslice-operator/pull/149/files

Comment (rphillips) [reply internal/controller/instaslice_controller.go]: code change looks fine just not go fmt'ed... Looks like the force push cleared it up.

**Comment by rphillips** [comment]
Should the hold be taken off of this PR?

**Comment by rphillips** [comment]
/override "Red Hat Konflux / dynamicacceleratorslicer-enterprise-contract / instaslice-operator-bundle"
/override "Red Hat Konflux / dynamicacceleratorslicer-enterprise-contract / instaslice-operator-controller"
/override "Red Hat Konflux / dynamicacceleratorslicer-enterprise-contract / instaslice-operator-daemonset"
/override "Red Hat Konflux / dynamicacceleratorslicer-enterprise-contract / pr group daemonset-installed-by-controller"

**Pull request merged**